osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -228,32 +229,41 @@ private async Task updatePlaylistOrder(IDatabaseAccess db)
                                                                 .First();
 
                         unprocessedItems.Remove(candidateItem);
-                        processedItems.Add(candidateItem);
+                        orderedActiveItems.Add(candidateItem);
 
                         perUserCounts[candidateItem.OwnerID]++;
                     }
 
-                    orderedItems = processedItems;
                     break;
             }
 
-            for (int i = 0; i < orderedItems.Count; i++)
+            // For expired items, it's important that they're ordered in ascending order such that the last updated item is the last in the list.
+            // This is so that the updated_at database column doesn't get refreshed as a result of change in ordering.
```

Comment:
This really sounds like you want another column that isn't `updated_at`. That's what we do in all other instances where we don't expect the column to be updated on write.

Actually important, since if laravel happens to touch one of these rows for any reason in the future, it *is* going to modify the column as it stands.

Reply:
Hmmm... So I guess we do need an `expired_at` anyway.

Reply:
I'd say that's safest, yeah.

Can happen as a separate change if that's easier.